src/eh1/delay.rs

```diff
@@ -68,12 +67,50 @@ impl delay::DelayNs for StdSleep {
     fn delay_ns(&mut self, ns: u32) {
         thread::sleep(Duration::from_nanos(ns as u64));
     }
+}
+
+#[cfg(feature = "embedded-hal-async")]
+impl embedded_hal_async::delay::DelayNs for StdSleep {
+    async fn delay_ns(&mut self, _ns: u32) {
+        // no-op
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
The tests below don't actually test anything, except that the functions don't panic. That's also, why the issue above wasn't caught.

Suggestion:

- In the noop tests, call `delay_ms(500)` and assert that less than 100 ms passed (by measuring time before and after)
- In std tests, call `delay_ms(100)` and assert that more than 100 ms passed (by measuring time before and after)

Use https://doc.rust-lang.org/std/time/struct.Instant.html as monotonic time source.